app/tools/android/test_data/extensions/echoJsStubGenExtension/src/com/example/jsStub/Echo.java

```diff
@@ -0,0 +1,141 @@
+package com.example.jsStub;
+
+import android.util.Log;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.xwalk.app.runtime.extension.*;
+
+/*
+ * Example for JavaScript stub auto-generation feature on Android.
+ * This extension will expose an object:
+ * {
+ *   readOnlyProperty: [string] read only
+ *   writablePrefix: [string] writable
+ *   testEvent: [Function], an async method
+ *   echo: [Function], a sync method
+ *   getPrifix: [Function], an async method with callback
+ *   getPrifixPromise: [Function], a method returns Promise
+ *
+ *   onupdatePrefix: [Function] writable, property of event handler entry
+ *   onclick: [Function] writable, property of event handler entry
+ *   onnewStudent: [Function] writable, property of event handler entry
+ *
+ *   addEventListener: [Function], interface method as an EventTarget
+ *   dispatchEvent: [Function], interface method as an EventTarget
+ *   removeEventListener: [Function], interface method as an EventTarget
+ * }
+ */
+public class Echo extends XWalkExtensionClient {
+
+  // Expose a read only JS property.
+  @JsApi
+  public String readOnlyProperty = "0x0000";
+
+  // Expose a writable JS property.
+  @JsApi(isWritable = true)
+  public String writablePrefix = "From Java:";
+
+  // The event list must be "static" field.
+  @JsApi(isEventList = true)
+  public static String[] events = {"updatePrefix", "click", "newStudent"};
+
+  public Echo(String extensionName, String jsApi, XWalkExtensionContextClient context) {
+    super(extensionName, jsApi, context);
+  }
+  
+  // Event object specified serializing method "toJSONObject", this must be public method.
+  public class Student {
+      private String name;
+      private int age;
+
+      public Student(String name, int age) {
+          this.name = name;
+          this.age = age;
+      }
+
+      // For Java side usage.
+      public String getName() {
+          return name;
+      }
+
+      // For Java side usage.
+      public int getAge() {
+          return age;
+      }
+
+      // The specified serialization method for JS side.
+      public String toJSONString() {
+          try {
+              JSONObject eventData = new JSONObject();
+              eventData.put("name", name);
+              eventData.put("age", age);
+              return eventData.toString();
+          } catch (Exception e) {
+              e.printStackTrace();
+              return "{}";
+          }
+      }
+  }
+
+  // Event object without customized serializing method.
```

Comment:
It will be serialized as a JSON string and passed to JS side, and properties in the parsed object from JSON is not read-only currently.